### DISCUSSION TITLE
A brand new admin theme for OpenMage

### DISCUSSION THREAD

**Discussion by empiricompany** [OpenMage/magento-lts#3458] [Show & Tell]
Hello everyone, I'm working on a new theme for the backend of OpenMage that aims to be more professional and organized. I must admit that I have never been liked the current OpenMage theme, and I provide my clients with a customized theme (different from this one) that is more practical and user-friendly.

So far, I have been working on the current theme, but I would like to rewrite it in a more organized manner. Since it would require time, I wanted to get some feedback to see if it's a project you would like me to continue and complete.

I have chosen neutral white and black as the main colors, with the primary OpenMage color used to highlight call-to-action elements and underline certain links to highlight secondary actions. This was also a strategic decision to avoid completely overhauling everything and ensure maximum backward compatibility with other extensions. Another strategic choice was to remove all the old PNG icons from the titles.

As you can see, everything has been enlarged to optimize it for more current resolutions than it was originally designed for, giving it a more modern and functional look with larger inputs and buttons.

It's not perfect, for example, the calendar, which now uses the primary color, needs to be revised, and there are certainly other small improvements to be made. But, personally, I really like it, it is more organized and there is a more appropriate use of spaces, dimensions and colors.

In my opinion, a more professional and user-friendly theme would help OpenMage gain recognition as a new good product in the open-source e-commerce market. This would distinguish it from the old Magento 1 and contribute to establishing its identity among vendors, agencies, and other key players.

I invite you to try it out, as there are many aspects of interaction with the GUI that cannot be highlighted by screenshots:
https://github.com/empiricompany/openmage/tree/backendtheme

This is a merged version with new tinymce 6 and advanced product grid:
https://github.com/empiricompany/openmage/tree/bestall

**Comment by empiricompany** [comment]
Click to enlarge screenshots:

![menu](https://github.com/OpenMage/magento-lts/assets/5071467/59047e96-afa2-4e88-bebf-597fa5498dd3)

![dashboard](https://github.com/OpenMage/magento-lts/assets/5071467/b4ee112b-6090-4df7-9d17-443f85e86d69)
(PS: here I need to replace the orange from the old Magento 1 with our main color)

![sales](https://github.com/OpenMage/magento-lts/assets/5071467/c7facfb0-0589-4aa0-875a-e3d3965590e4)
![order_view](https://github.com/OpenMage/magento-lts/assets/5071467/c4a428fe-0259-4309-a557-70e0d397e6cd)

![category](https://github.com/OpenMage/magento-lts/assets/5071467/9ff7ecd3-f95a-4b36-b608-9cf8b40f9e9c)

![catalog_product](https://github.com/OpenMage/magento-lts/assets/5071467/ddc612e4-7f2f-4f04-ac6a-af9001d74407)
![product_edit](https://github.com/OpenMage/magento-lts/assets/5071467/081f55ac-7f6b-4817-80c1-6b0f5ce1d42b)

![cms](https://github.com/OpenMage/magento-lts/assets/5071467/37374cee-765c-4b79-a497-3272c1690289)

![review](https://github.com/OpenMage/magento-lts/assets/5071467/084c2063-9f61-4a6f-b35d-a98662ab3dad)

![cache](https://github.com/OpenMage/magento-lts/assets/5071467/d58916cb-d82c-4e66-a555-c913c99e3564)
![indexes](https://github.com/OpenMage/magento-lts/assets/5071467/f126925b-eb19-41e0-a863-f0348a4a365d)

![reports](https://github.com/OpenMage/magento-lts/assets/5071467/37852376-5aec-47a0-b1eb-4e08a780f779)

![system_config](https://github.com/OpenMage/magento-lts/assets/5071467/eadddeee-3b52-429e-b359-d9b96454b27d)

**Comment by empiricompany** [comment]
this is a version less disruptive with the same colors
![openmage_new_theme_legacy_colors](https://github.com/OpenMage/magento-lts/assets/5071467/2bb18006-56e1-4b4f-a02d-ff7aaa11c0ed)

https://github.com/empiricompany/openmage/tree/backendtheme_main

the menu and buttons colors are less important then others adjustment i made in the ui (spaces, alignments, inputs etc..)

**Reply by elidrissidev** [reply]
I like this one a lot! the current OM theme lacks contrast in many places.

**Reply by fballiano** [reply]
I also like this one, the difference with the current one are just the colors a the rounded corners of the buttons?

**Reply by empiricompany** [reply]
@fballiano no there are many little things in spaces, input, alignments...and other described in head of this PR...needs to try it to capture all the differences

**Reply by empiricompany** [reply]
updated https://github.com/empiricompany/openmage/tree/backendtheme_main
with current color schema and a bit of purple of our brand color for highlights
![legacy_colors](https://github.com/OpenMage/magento-lts/assets/5071467/27621268-6ebb-4b2f-bd5e-7ec40b899529)

**Comment by empiricompany** [comment]
another good combination
![other_combinations](https://github.com/OpenMage/magento-lts/assets/5071467/d75b9fdf-5523-48df-b257-d317dcc16df7)
![other_combinations2](https://github.com/OpenMage/magento-lts/assets/5071467/10fc72fb-3b4e-48ca-842f-922749508d5d)

combinations are endless IMHO it's important to mantain consistency with our brand identity

**Comment by Danieleeffe1** [comment]
nice work, if you are interested I just published my MIMIC frontend theme.

I'm not very used to teamwork and will create better descriptions and instructions with time.

If you are interested in helping me develop it, here you go:

https://github.com/Danieleeffe1/OpenMage-Themes/tree/main

**Comment by LuizSantos22** [comment]
@empiricompany 
Wow that really shines... 
Here in Brazil there's a private company called "Inovarti" which bases their platform in OpenMage
I used to hate vertical menu, but what they did is outstanding...
It's a pitty that we do not gain access to the source code, but here you have a link with the screenshots: 
https://www.inovarti.com.br/inovarti-platform-e-atualizacoes-para-2021/

**Reply by fballiano** [reply]
did they ever participate in openmage? are they sponsoring somebody here?

**Reply by LuizSantos22** [reply]
@fballiano I do not have this answer, the responsible of the company used to contribute with some extensions on Github. Like the OneStepCheckout to Brazil market which was iniatially based on IWD OPC with adaptations to our market.

**Comment by empiricompany** [comment]
@LuizSantos22 -  Even though it looks very nice from the screenshots, an extreme customization like that with a vertical menu could cause compatibility problems with third-party extensions. Our intention here is different; we simply want to update and improve the current theme with some minor adjustments to make it appear more professional and user-friendly.

But regarding the inovarti fork, if it's a fork of the core, they should release the source code of their platform; otherwise, they would not be respecting the OpenMage license in my opinion. I do something similar to add some features, but it's all developed through extensions and server-side patches, and I don't think they do the same.

**Comment by LuizSantos22** [comment]
@empiricompany 
Hi there, their frontend and graphic looks are based on Bootstrap, probably v4. 
As they sell the "service" (it acts like SAAS) I do not think they would open
the source code of the whole thing and actually, most of the changes are "extensions"
installed on the core (OpenMage).

In terms of compability issues, I do not think it causes more issues than any other
extension. Bootstrap (I am not an expert, just used it once in a while) is quite
lightweight and fixes a lot of issues just using grids and tiny snippet codes to call icons
or actions... So it is quite clean to work with and avoid a plethora of compability issues with
other extensions as it strips off a bunch of javascripts or jquery call used in the past to get same
effect.

Anyway, I will try your approach later, in my production I use TIG Admin theme but it is 
quite old!:

**Reply by empiricompany** [reply]
if you modify the HTML structure or change the classes, as I believe they have done, it's normal to create incompatibilities with some extensions that expect a different structure.

Regarding the license i don't think they have modified it through extensions but it should be legitimate (if it's not a SaaS but you can download and copy the entire source code). We are discussing it here:
https://github.com/OpenMage/magento-lts/discussions/3581

**Comment by LuizSantos22** [comment]
@empiricompany 
It might, it does not mean it will. For instance, TIG uses vertical menu and it has more potential to have conflicts as it uses old technologies and surprisingly it doesn't have compability issues with any of my extensions and I have installed a lot of them, even old ones to test it before applying on production. 
Anyway, I am seeing the discussions about the licence... It would be fun seeing their project on Github, but I don't think they will bend over that easily.  

And yes, Inovarti works as SAAS, quite expensive, actually. Their lowest plan is about BRL65k (over USD12.5k) plus monthly fees...
They are more focused on mid range to big companies here in Brazil. 


Here is the plan link: 
https://www.inovarti.com.br/planos

**Comment by LuizSantos22** [comment]
@empiricompany Do you have this admin themes as standalone extensions? I am trying to find them but I couldn't 
I would like to test them...

Regards

**Reply by empiricompany** [reply]
you can clone this repo:
https://github.com/empiricompany/openmage/tree/backendtheme

then copy:
app/design/adminhtml/default/openmage
skin/adminhtml/default/openmage

**Reply by LuizSantos22** [reply]
@empiricompany 
Thanks, but I will try, my suggestion would create it as an extension so when installed the customer can change it or revert back to default... 
Anyway, I did not try it yet so I don't know if yours works like that...

But think about it as an extension like [jhejlik/Mageek_OpenmageTheme you've recommended below (I used it in some projects).

Thank you once again.

**Comment by empiricompany** [comment]
I also recommend this theme by @jhejlik: 
https://github.com/jhejlik/Mageek_OpenmageTheme

**Reply by empiricompany** [reply]
But it seems more invasive, overwriting several files

**Comment by kiatng** [comment]
How about theme fallback? The backend admin can have similar fallback like the frontend. For this, we need to move the current default folder to base.  From here `app/design/adminhtml/default/default` to here `app/design/adminhtml/base/default`, same with the skin folder. This makes it easier to customize the core phtml files and also to have different themes such as proposed in this discussion.

**Reply by elidrissidev** [reply]
Exactly, that will also fix #1216 I believe.

**Comment by ma4nn** [comment]
Me personally - for nostalgic reasons - I still stick with the default (old-school) Magento Admin theme in all maintained installations, so please don't remove it😆

**Comment by empiricompany** [comment]
@fballiano @Flyingmana @kiatng @sreichel (and all mantainers / supporters) please let's discuss about this. 
I'd like to come out with a renewed and modern theme in the release 21 to make the difference and attract new users

**Reply by Flyingmana** [reply]
if its as a separate theme, not affecting the current one, you get a clear approve from my side, else I would need to put more time into reviewing details.
(although, as long as it has enough support you can also proceed without my approval)

**Reply by Danieleeffe1** [reply]
I purchased an adaptive theme for Magento 1 which however gives problems on openmage <20.05.

The site where I purchased it hasn't been active for years and I can't find the developer.

Could that be used as a basis or inspiration and adapted to OM 21?

**Reply by empiricompany** [reply]
they are only small changes to the current OpenMage backend theme.
if we agree all I can take the time to create an official PR with only the changes.
there are for now two versions of color schema:
- https://github.com/empiricompany/openmage/tree/backendtheme
- https://github.com/empiricompany/openmage/tree/backendtheme_main

**Reply by empiricompany** [reply]
@Danieleeffe1 no thanks as i said this is only a revamped version of current backend theme of OpenMage with small changes

**Reply by fballiano** [reply]
it's also very tricky to take inspiration from copyrighted code.

at the moment I'd like (just talking about what I personally can focus on) to reduce the number of current PRs and get to merge what we have already and that still needs tests or small fixes :-) but anyway why not.

personally i didn't love the black version (from the old screenshots), I think maybe the black main menu, but now I don't remember everything

**Comment by jparker1986** [comment]
Personally feel the black theme says WordPress more than OpenMage.

Current theme does not bother me. I actually think despite it's long out dated design it works really well.  

Only gripe with the current theme is the menus are not mobile friendly, and with most admins working on the go it's certainly more tricky to navigate than other platforms. 

If there is a sub menu in a menu (Sales / Catalogue etc) you have to be quick to select one otherwise the page reloads before you can click anything. A truck I discovered is a long hold on the menu will open it but not reload the page and enable you to select sub menus.

**Comment by empiricompany** [comment]
We need a breath of freshness; we need to renew ourselves. If we remain anchored to the past, this project can only die. 
That's why I think we should eliminate even the original Magento theme and any other reference.
IMHO The current theme has a dated design and makes the product appear old, failing to showcase all the new features introduced in version 21

**Reply by jparker1986** [reply]
Yes I do agree with most of the points you raised but disagree the project will simply die because of a dated backend.

As a store owner, I would much prefer time and energy goes into making the project secure and working with latest versions of PHP.

The theme you showed above:

https://github.com/jhejlik/Mageek_OpenmageTheme

Is actually very good. Retains the classic feel of OM and also gives it a gentle refresh.

We have many third party systems which integrate with Magento via db and APIs ( all build in Laravel ) which give us a nodern front end In any case, but the most of the layout we copied from OM as it's very easy to find information on the screen.

**Reply by empiricompany** [reply]
The theme you mentioned was the starting point, and now it's completely different from the current one.

If it were like that, it would be much better! 
![originaltheme](https://github.com/OpenMage/magento-lts/assets/5071467/40b58892-4e31-41a2-a707-116e73ebb033)

But it's overly personalized and could cause issues with various third-party extensions. 
The current theme is super-compatible because it doesn't overhaul the original design, and with some minor tweaks, it can be improved to make it more modern and user-friendly.

IMHO more decisive choices need to be made if we want to move forward. OpenMage is now a new platform, and while maintaining as much backward compatibility as possible, need to be competitive with other platforms. 
My vision of the future of OpenMage is very different from many of you who are quite conservative. 
In fact, I would like to create a fork like 'OpenMage Plus,' but I wouldn't be able to manage it alone. 🤔

PS: This is just my opinion, I don't want to impose anything.

**Reply by jparker1986** [reply]
I think we are already at that stage now. The current beta version that Fab released the other day has a huge exodus of files deleted that have lingered in the file base since day 1. 

In order to improve the house, you must first clean it of the clutter. 

The problem with the above screenshot you sent is it's just a bit of a merge between what could be and what is now. For me the colours are too much.

**Reply by Flyingmana** [reply]
Iam not following the topic very closely, but 2 opinions from my side.

> We need a breath of freshness; we need to renew ourselves. If we remain anchored to the past, this project can only die.
> That's why I think we should eliminate even the original Magento theme and any other reference.
> IMHO The current theme has a dated design and makes the product appear old, failing to showcase all the new features introduced in version 21

We should not delete anything without a full functional and also proven alternative. Creating a new Theme and directly deleting the old one is what is killing a Project, because it shows to not be a reliable. So please whatever you do in this regard, calculate in a grace period of at least one major release.
(please be more careful and avoid using fear mongering as method to pressure certain changes)


> The theme you mentioned was the starting point, and now it's completely different from the current one.
> 
> If it were like that, it would be much better! ![originaltheme](https://private-user-images.githubusercontent.com/5071467/322207856-40b58892-4e31-41a2-a707-116e73ebb033.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMwMjA3OTgsIm5iZiI6MTcxMzAyMDQ5OCwicGF0aCI6Ii81MDcxNDY3LzMyMjIwNzg1Ni00MGI1ODg5Mi00ZTMxLTQxYTItYTcwNy0xMTZlNzNlYmIwMzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTNUMTUwMTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODBmNDBjNWY1OWJmYWMyMGUxYWIyYjk3NzExNDk1M2UxM2M1NGE0NTZlMjBmYzQxYjk1ZWRkNjMwYjg4ZjhjZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.u53oPvmsiMA9-Kkct11WLA4ZJMrwHEq4DOn3z8bxAmk)
> 

Not sure where this screenshot is from, but it seems not taking into account accessibility issues like color blindness. Also the contrast seems seems to be off a bit. (just meant as a hint, not a complaint)

**Reply by empiricompany** [reply]
@Flyingmana not fear here only to say OpenMage is not Magento 1.9 anymore and I think we have to start to say it as product, for me.
